sdk/python/toolchain/poetry.go

```diff
@@ -151,3 +167,107 @@ func (p *poetry) virtualenvPath(ctx context.Context) (string, error) {
 	}
 	return virtualenvPath, nil
 }
+
+func (p *poetry) convertRequirementsTxt(requirementsTxt, pyprojectToml string) error {
+	f, err := os.Open(requirementsTxt)
+	if err != nil {
+		return fmt.Errorf("failed to open %q", requirementsTxt)
+	}
+	defer f.Close()
+
+	deps, err := dependenciesFromRequirementsTxt(f)
+	if err != nil {
+		return fmt.Errorf("failed to gather dependencies from %q", requirementsTxt)
+	}
+
+	b, err := p.generatePyProjectTOML(deps)
+	if err != nil {
+		return fmt.Errorf("failed to generate %q", pyprojectToml)
+	}
+
+	pyprojectFile, err := os.Create(pyprojectToml)
+	if err != nil {
+		return fmt.Errorf("failed to create %q", pyprojectToml)
+	}
+	defer pyprojectFile.Close()
+
+	if _, err := pyprojectFile.Write([]byte(b)); err != nil {
+		return fmt.Errorf("failed to write to %q", pyprojectToml)
+	}
+
+	if err := os.Remove(requirementsTxt); err != nil {
+		return fmt.Errorf("failed to remove %q", requirementsTxt)
+	}
+
+	return nil
+}
+
+func (p *poetry) generatePyProjectTOML(dependencies map[string]string) (string, error) {
+	type BuildSystem struct {
+		Requires     []string `toml:"requires,omitempty" json:"requires,omitempty"`
+		BuildBackend string   `toml:"build-backend,omitempty" json:"build-backend,omitempty"`
+	}
+
+	type Pyproject struct {
+		BuildSystem *BuildSystem           `toml:"build-system,omitempty" json:"build-system,omitempty"`
+		Tool        map[string]interface{} `toml:"tool,omitempty" json:"tool,omitempty"`
+	}
+
+	pp := Pyproject{
+		BuildSystem: &BuildSystem{
+			Requires:     []string{"poetry-core"},
+			BuildBackend: "poetry.core.masonry.api",
+		},
+		Tool: map[string]any{
+			"poetry": map[string]any{
+				"package-mode": false,
+				"dependencies": dependencies,
+			},
+		},
+	}
+
+	w := &bytes.Buffer{}
+	if err := toml.NewEncoder(w).Encode(pp); err != nil {
+		return "", err
+	}
+	return w.String(), nil
+}
+
+func dependenciesFromRequirementsTxt(r io.Reader) (map[string]string, error) {
+	versionRe := regexp.MustCompile("[<>=]+.+")
```

Comment:
Will this work for operators like `~=`?

https://peps.python.org/pep-0508/#grammar